test/recipes/04-test_pem.t

```diff
@@ -0,0 +1,102 @@
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+# ======================================================================
+
+
+use strict;
+use warnings;
+
+use File::Compare qw/compare_text/;
+use File::Basename;
+use OpenSSL::Test qw/:DEFAULT srctop_file data_file/;
+use OpenSSL::Test::Utils;
+
+setup("test_pem_reading");
+
+my $testsrc = srctop_file("test", "recipes", basename($0));
+
+my $cmd = "openssl";
+
+# map input PEM file to 1 if it should be accepted; 0 when should be rejected
+my %cert_expected = (
+    "cert-1023line.pem" => 1,
+    "cert-1024line.pem" => 1,
+    "cert-1025line.pem" => 1,
+    "cert-255line.pem" => 1,
+    "cert-256line.pem" => 1,
+    "cert-257line.pem" => 1,
+    "cert-blankline.pem" => 0,
+    "cert-comment.pem" => 0,
+    "cert-earlypad.pem" => 0,
+    "cert-extrapad.pem" => 0,
+    "cert-infixwhitespace.pem" => 1,
+    "cert-junk.pem" => 0,
+    "cert-leadingwhitespace.pem" => 1,
+    "cert-longline.pem" => 1,
+    "cert-misalignedpad.pem" => 0,
+    "cert-onecolumn.pem" => 1,
+    "cert-oneline.pem" => 1,
+    "cert-shortandlongline.pem" => 1,
+    "cert-shortline.pem" => 1,
+    "cert-threecolumn.pem" => 1,
+    "cert-trailingwhitespace.pem" => 1,
+    "cert.pem" => 1
+);
+my %dsa_expected = (
+    "dsa-1023line.pem" => 0,
+    "dsa-1024line.pem" => 0,
+    "dsa-1025line.pem" => 0,
+    "dsa-255line.pem" => 0,
+    "dsa-256line.pem" => 0,
+    "dsa-257line.pem" => 0,
+    "dsa-blankline.pem" => 0,
+    "dsa-comment.pem" => 0,
+    "dsa-corruptedheader.pem" => 0,
+    "dsa-corruptiv.pem" => 0,
+    "dsa-earlypad.pem" => 0,
+    "dsa-extrapad.pem" => 0,
+    "dsa-infixwhitespace.pem" => 0,
+    "dsa-junk.pem" => 0,
+    "dsa-leadingwhitespace.pem" => 0,
+    "dsa-longline.pem" => 0,
+    "dsa-misalignedpad.pem" => 0,
+    "dsa-onecolumn.pem" => 0,
+    "dsa-oneline.pem" => 0,
+    "dsa-onelineheader.pem" => 0,
+    "dsa-shortandlongline.pem" => 0,
+    "dsa-shortline.pem" => 0,
+    "dsa-threecolumn.pem" => 0,
+    "dsa-trailingwhitespace.pem" => 1,
+    "dsa.pem" => 1
+);
+
+plan tests =>  scalar keys(%cert_expected) + scalar keys(%dsa_expected) + 1;
+
+SKIP: {
+    #"agl.pem" => 1,
+    #"beermug.pem" => 1,
+    foreach my $input (keys %cert_expected) {
+        my @common = ($cmd, "x509", "-text", "-noout", "-inform", "PEM", "-in");
+        my @data = run(app([@common, data_file($input)], stderr => undef), capture => 1);
+        my @match = grep /The Great State of Long-Winded Certificate Field Names Whereby to Increase the Output Size/, @data;
+        is((scalar @match > 0 ? 1 : 0), $cert_expected{$input});
+    }
+    foreach my $input (keys %dsa_expected) {
+        my @common = ($cmd, "pkey", "-inform", "PEM", "-passin", "file:" . data_file("wellknown"), "-noout", "-text", "-in");
+        skip "DSA support disabled, skipping...", (scalar keys %dsa_expected) unless !disabled("dsa");
```

Comment:
Ah, this line should be outside the `for` statement, not inside.  Also, did you know that you can repeat the same label again and again?  They are only tied to the block that follows them.  So!  you should remove the `SKIP:` you currently have, and...  Oh, let me show you by example:

``` perl
#"agl.pem" => 1,
#"beermug.pem" => 1,
foreach my $input (keys %cert_expected) {
    # ... body ...
}
SKIP: {
    skip "DSA support disabled, skipping...", (scalar keys %dsa_expected) unless !disabled("dsa");
    foreach my $input (keys %dsa_expected) {
        # ... body ...
    }
}
SKIP: {
    skip "RSA support disabled, skipping...", 1 unless !disabled("rsa");
    my @common = # ... etc etc etc
    # ... etc etc etc
}
```